alkira/resource_alkira_connector_azure_er.go

```diff
@@ -0,0 +1,294 @@
+							Required: true,
+						},
+						"redundant_router": {
+							Description: "Indicates if ExpressRoute Circuit terminates on redundant routers on customer side.",
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Default:     false,
+						},
+						"loopback_subnet": {
+							Description: "A `/26` subnet from which loopback IPs would be used to establish underlay VXLAN GPE tunnels.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"credential_id": {
+							Description: "An opaque identifier generated when storing Azure VNET credentials.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"gateway_mac_address": {
+							Description: "An array containing the mac addresses of VXLAN gateways reachable through ExpressRoute circuit. " +
+								"The gatewayMacAddresses is only expected if VXLAN tunnel protocol is selected, " +
+								"and 2 gateway mac addresses are expected only if redundant_router is enabled.",
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"virtual_network_interface": {
+							Description: "This is an optional field if the tunnel protocol is VXLAN. " +
+								"If not specified Alkira allocates unique VNI from the range [16773023, 16777215].",
+							Type:     schema.TypeList,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeInt},
+						},
+					},
+				},
+			},
+			"segment_options": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"segment_name": {
+							Description: "The name of an existing segment.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"segment_id": {
+							Description: "The ID of the segment.",
+							Type:        schema.TypeInt,
+							Computed:    true,
+						},
+						"customer_asn": {
+							Description: "ASN on the customer premise side.",
+							Type:        schema.TypeInt,
+							Required:    true,
+						},
+						"disable_internet_exit": {
+							Description: "Enable or disable access to the internet when traffic arrives via this connector.",
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Default:     true,
+						},
+						"advertise_on_prem_routes": {
+							Description: "Allow routes from the branch/premises to be advertised to the cloud.",
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Default:     false,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// resourceConnectorAzureErCreate create an Azure ExpressRoute connector
+func resourceConnectorAzureErCreate(d *schema.ResourceData, m interface{}) error {
+	client := m.(*alkira.AlkiraClient)
+	connector, err := generateConnectorAzureErRequest(d, m)
+
+	if err != nil {
+		return err
+	}
+
+	id, err := client.CreateConnectorAzureEr(connector)
```

Comment:
The client change was not here yet? Guess the original function name is a bit confusing...